enterprise/backend/src/metabase_enterprise/caching/strategies.clj

```diff
@@ -0,0 +1,114 @@
+(ns metabase-enterprise.caching.strategies
+  (:require
+   [java-time.api :as t]
+   [malli.core :as mc]
+   [metabase.public-settings :as public-settings]
+   [metabase.public-settings.premium-features :refer [defenterprise defenterprise-schema]]
+   [metabase.query-processor.middleware.cache-backend.db :as backend.db]
+   [metabase.util.cron :as u.cron]
+   [metabase.util.log :as log]
+   [metabase.util.malli.schema :as ms]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+(def CacheStrategy ;; not used in API since there are some additional fields
+  "Schema for a caching strategy"
+  (mc/schema
+   [:and
+    [:map
+     [:type [:enum :nocache :ttl :duration :schedule :query]]]
+    [:multi {:dispatch :type}
+     [:nocache  [:map
+                 [:type keyword?]]]
+     [:ttl      [:map
+                 [:type keyword?]
+                 [:multiplier ms/PositiveInt]
+                 [:min_duration ms/PositiveInt]]]
+     [:duration [:map
+                 [:type keyword?]
+                 [:duration ms/PositiveInt]
+                 [:unit [:enum "hours" "minutes" "seconds" "days"]]]]
+     [:schedule [:map
+                 [:type keyword?]
+                 [:last-expired-at [:maybe some?]]
+                 [:schedule u.cron/CronScheduleString]]]
+     [:query    [:map
```

Comment:
this is missing `table_id`

Reply:
I'm leaning towards `{:closed true}` schemas here 😄

Reply:
Done except for `nocache` - I'm abusing it in tests to see that correct values are being used. Not sure what to do with that. 😁